app/code/Magento/Directory/etc/zip_codes.xml

```diff
@@ -258,7 +258,8 @@
     </zip>
     <zip countryCode="GG">
         <codes>
-            <code id="pattern_1" active="true" example="AB1 2CD">^[a-zA-Z]{2}[0-9]{1}\s[0-9]{1}[a-zA-Z]{2}$</code>
+            <code id="pattern_1" active="true" example="GY1 2AB">^GY[0-9]{1}\s[0-9]{1}[a-zA-Z]{2}$</code>
+            <code id="pattern_2" active="true" example="GY10 1PB">^GY[0-9]{2}\s?[0-9]{1}[a-zA-Z]{2}$</code>
```

Comment:
Just expand a single pattern instead of adding a new one (`GY[0-9]{1}` -> `GY[0-9]{2}`).

Would be _really_ nice to cover this revealed case with a unit test.

After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉

Reply:
> Just expand a single pattern instead of adding a new one (`GY[0-9]{1}` -> `GY[0-9]{2}`).

@orlangur postcode could have digit from `0-9` or `10`.

I suggest another pattern `^GY([0-9]|10)\s[0-9][a-zA-Z]{2}`

Reply:
Applied updates via https://github.com/magento/magento2/pull/33128/commits/d2e11d06acd169ff3ba6a9ff06da1a3eba26dc05

I also fixed another bug with Jersey postcodes if customers enter postcodes without spaces.